src/Symfony/Bundle/FrameworkBundle/Resources/config/templating_php.xml

```diff
@@ -11,71 +11,99 @@
         </service>
 
         <service id="templating.helper.code" class="Symfony\Bundle\FrameworkBundle\Templating\Helper\CodeHelper">
             <tag name="templating.helper" alias="code" />
             <argument type="service" id="debug.file_link_formatter"></argument>
             <argument>%kernel.root_dir%</argument>
             <argument>%kernel.charset%</argument>
+
+            <deprecated>The "%service_id%" service is deprecated since Symfony 3.3 and will be removed in 4.0.</deprecated>
         </service>
 
         <service id="templating.helper.translator" class="Symfony\Bundle\FrameworkBundle\Templating\Helper\TranslatorHelper">
             <tag name="templating.helper" alias="translator" />
             <argument type="service" id="translator" />
+
+            <deprecated>The "%service_id%" service is deprecated since Symfony 3.3 and will be removed in 4.0.</deprecated>
         </service>
 
         <service id="templating.helper.form" class="Symfony\Bundle\FrameworkBundle\Templating\Helper\FormHelper">
             <tag name="templating.helper" alias="form" />
             <argument type="service" id="templating.form.renderer" />
+
+            <deprecated>The "%service_id%" service is deprecated since Symfony 3.3 and will be removed in 4.0.</deprecated>
         </service>
 
         <service id="templating.helper.stopwatch" class="Symfony\Bundle\FrameworkBundle\Templating\Helper\StopwatchHelper">
             <tag name="templating.helper" alias="stopwatch" />
             <argument type="service" id="debug.stopwatch" on-invalid="ignore" />
+
+            <deprecated>The "%service_id%" service is deprecated since Symfony 3.3 and will be removed in 4.0.</deprecated>
         </service>
 
         <service id="templating.form.engine" class="Symfony\Component\Form\Extension\Templating\TemplatingRendererEngine" public="false">
             <argument type="service" id="templating.engine.php" />
             <argument>%templating.helper.form.resources%</argument>
+
+            <deprecated>The "%service_id%" service is deprecated since Symfony 3.3 and will be removed in 4.0.</deprecated>
         </service>
 
         <service id="templating.form.renderer" class="Symfony\Component\Form\FormRenderer" public="false">
             <argument type="service" id="templating.form.engine" />
             <argument type="service" id="security.csrf.token_manager" on-invalid="null" />
+
+            <deprecated>The "%service_id%" service is deprecated since Symfony 3.3 and will be removed in 4.0.</deprecated>
         </service>
 
         <service id="templating.globals" class="Symfony\Bundle\FrameworkBundle\Templating\GlobalVariables">
             <argument type="service" id="service_container" />
+
+            <deprecated>The "%service_id%" service is deprecated since Symfony 3.3 and will be removed in 4.0.</deprecated>
```

Comment:
could we use `_defaults` instead of updating them all? (would apply to other files)

Reply:
_defaults doesn't support the `deprecated` attribute.